metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/bolt/ParserBolt.java

```diff
@@ -60,7 +93,25 @@ public void prepare(Map stormConf, TopologyContext context, OutputCollector coll
     super.prepare(stormConf, context, collector);
     this.collector = collector;
     parser.init();
-    writer.init();
+
+    if(isBulk) {
+      writerTransformer = config -> new ParserWriterConfiguration(config);
```

Comment:
Curious why a Function is used here.  Why not just instantiate a WriterConfiguration object for each case and pass that to messageWriter.init?  Is there a benefit to doing it this way?